### PR TITLE
Update: Set property to change one property accept all lodash paths.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
When trying to address https://github.com/WordPress/gutenberg/pull/25145#issuecomment-688823538, I noticed our setProperty global styles methods do not accept every lodash path but only string paths.
Previously we discovered that using string paths is slow and we have a lint rule that disallows string paths on lodash e.g: `get ( object, 'a.b' )` is not accepted by our lint rules.
I guess we should use array paths like the rest of the codebase using array paths would also allow us to use the global mappings that are in an array format.

The setProperty method allows changing multiple properties, each property is a key in the object, we can not use keys that are arrays. So this PR changes setProperty to change a single property at a time.

## How has this been tested?
I verified the global styles sidebar continues to work.

### CODE REVIEW

**Comment by jorgefilipecosta** [comment]
This PR seems to fix the custom gradient problem, so I enabled them again.

File: packages/edit-site/src/components/sidebar/color-panel.js [hunk 1]
```diff
@@ -7,12 +7,11 @@ import { __ } from '@wordpress/i18n';
...
  */
-import {
-	BACKGROUND_COLOR,
-	LINK_COLOR,
-	TEXT_COLOR,
-	GRADIENT_COLOR,
-} from '../editor/utils';
+import { LINK_COLOR } from '../editor/utils';
+
+const BACKGROUND_COLOR = 'background-color';
```
Comment (nosolosw) [inline packages/edit-site/src/components/sidebar/color-panel.js:12]: I'd rather keep these are they are and we can discuss how to solve it in the other PR (I'm still unconvinced). This PR is so great that it merits landing asap.

**Review by oandregal — APPROVED** [review]
This is so great to see that I went ahead and pushed a couple of minor tweaks, so we can expedite this and merge as soon as CI gives us a :green_circle: 

To be honest, I tried this _same_ approach and it didn't work (obviously it wasn't the same, otherwise it _will_ work). I didn't like the promise trick to coordinate gradient/background setters but failed to make it work without it (the two consecutive setters will interrupt each other and the gradients panel didn't work). I'm very happy to see this fixed!

**Comment by nosolosw** [comment]
> Previously we discovered that using string paths is slow and we have a lint rule that disallows string paths on lodash e.g: get ( object, 'a.b' ) is not accepted by our lint rules.

This is a surprise to me. We do use string paths in a few places at the moment and there's no rule for this (although there [was one](https://github.com/WordPress/gutenberg/pull/6247) a couple of years ago, but can't find when/why it was removed). The comment about adding the rule reported that the performance gains were minimal. I'm not too opinionated about one form or another, but if committing to one makes things consistent through the codebase, I'm game for that change.

**Comment by jorgefilipecosta** [comment]
> This is a surprise to me. We do use string paths in a few places at the moment and there's no rule for this (although there was one a couple of years ago, but can't find when/why it was removed). The comment about adding the rule reported that the performance gains were minimal.

I was not aware the rule was removed and I thought we still did not use string paths. Anyway, the change made does not block string paths, it makes it possible to use any path format, while before only string paths worked.

**Pull request merged**